example/glue/EvacuatorDelegate.hpp

```diff
@@ -0,0 +1,245 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef EVACUATORDELEGATE_HPP_
+#define EVACUATORDELEGATE_HPP_
+
+#include "omr.h"
+#include "omrcfg.h"
+#include "omrExampleVM.hpp"
+#include "omrhashtable.h"
+
+#include "EvacuatorBase.hpp"
+#include "EnvironmentStandard.hpp"
+#include "ForwardedHeader.hpp"
+#include "IndexableObjectScanner.hpp"
+#include "MixedObjectScanner.hpp"
+
+class MM_Evacuator;
+class MM_EvacuatorController;
+class MM_Scavenger;
+
+class MM_EvacuatorDelegate
+{
+/*
+ * Data members
+ */
+public:
+	/**
+	 * Controller maintains a volatile bitset of delegate flags, evacuators mutate and test
+	 * these to control evacuation stages of each gc cycle. Evacuator delegate may define
+	 * flags as required to share this information across evacuators during gc cycles.
+	 *
+	 * @see MM_EvacuatorController::max_evacuator_public_flag (public bit mask is 0xffffffff)
+	 */
+	enum {
+		unused = 1
+	};
+private:
+	MM_EnvironmentStandard *_env;
+	MM_Scavenger *_controller;
+	MM_Evacuator *_evacuator;
+	bool _isCleared;
```

Comment:
Can you document the instance variables please?

Reply:
Will do.

Reply:
Done. Also removed redundant MM_EvacuatorDelegate::_env field and updated method comments to explain the persistence of the evacuator-delegate binding (1-1, invariant over delegate lifetime) and environment-(evacuator-delegate) binding (1-(1-1), invariant over duration of gc cycle, dissolved between gc cycles)